docs/contributors/write-documentation.rst

```diff
@@ -0,0 +1,185 @@
+
+There are two major types of documentation:
+
+1. **docstrings**: your code's docstrings will be automatically parsed by the documentation sofware (`Sphinx <https://www.sphinx-doc.org>`_, more in `about shpinx`_).
+2. **Manual** documentation such as this document.
+
+**Both** are written in `ReStructured Text <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_ (``.rst``) format.
+
+Some of the great features of using Sphinx is to be able to automatically generate documentation from your code's docstrings, and to be able to link to other parts of the documentation.
+
+For instance: :meth:`~gflownet.gflownet.GFlowNetAgent.trajectorybalance_loss` or to an external function :func:`torch.cuda.synchronize()`.
+
+.. caution::
+
+    This page is about learning the syntax of ``.rst`` and Sphinx. It is not about learning how to write good, informative docstrings.
+    Refer to `Google's docstring guide <https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings>`_ for that, or
+    checkout full-fleshed examples in `Numpy style <https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#example-numpy>`_ or `Google style <https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google>`_. We also have a short intro in the :doc:`next section </contributors/write-docstrings>`.
+
+Learn by example
+^^^^^^^^^^^^^^^^
+
+The next section will introduce many of the cool features of ``.rst`` + Sphinx + plugins.
+
+Click on "*Code for the example*" to look at the ``.rst`` code that generated what you are reading.
+
+.. tab-set::
+
+    .. tab-item:: Full-fleshed ``.rst`` example
+
+        .. include:: example.rst
+
+    .. tab-item:: Code for the example
+
+        .. literalinclude:: example.rst
+            :language: rst
+
+FAQ
+---
+
+.. dropdown:: How do I create new manual documentation files.
+
+    - Create a new ``.rst`` file in the ``docs/`` folder
+    - List it in ``docs/index.rst`` file under the ``.. toctree::`` directive
+    - **Or** create a subfolder in ``docs/`` with an ``index.rst`` file.
+        - This is useful for grouping documentation files together.
+        - ``docs/{your_subfolder}/index.rst`` should contain a ``.. toctree::`` directive listing the files in the subfolder.
+        - It should also be listed in the ``docs/index.rst`` under the ``.. toctree::`` directive to appear on the left handside of the documentation.
+
+    You can look at the |contributors|_ folder for an example.
+
+.. dropdown:: How do I document a sub-package like :py:mod:`gflownet.proxy.crystals`?
+
+    Just add a docstring at the top of the ``__init__.py`` file of the sub-package:
+
+    .. code-block:: python
+
+        """
+        This is the docstring of the sub-package.
+
+        It can contain any kind of ``.rst`` syntax.
+
+        And refer to its members: :meth:`~gflownet.proxy.crystals.crystal.Stage`
+
+        .. note::
+            This is a note admonition.
```

Comment:
Don't you need an empty line between `.. note::` and the content? (as per a note above)

Reply:
Yes typo

Reply:
Told you, common mistake ..